VAMobile/src/utils/waygateConfig.ts

```diff
@@ -0,0 +1,170 @@
+  WG_WhatToKnowScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_ContactInformationScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_HowWillYouScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_EditAddressScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_EditPhoneNumberScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_EditEmailScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_MilitaryInformationScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_IncorrectServiceInfoScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_SettingsScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_ManageYourAccountScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_NotificationsSettingsScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_ContactVAScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_VeteransCrisisLineScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_VeteranStatusScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_LoginScreen: JSON.parse(JSON.stringify(waygateDefault)),
+}
+
+export const waygateEnabled = (feature: WaygateToggleType): Waygate => {
+  if (overrideRemote) {
+    return waygateConfig[feature]
+  } else {
+    const waygate = remoteConfig().getValue(feature)?.asString()
+    if (waygate) {
+      return JSON.parse(waygate) as Waygate
+    } else {
+      return {
+        enabled: true,
+        errorMsgTitle: undefined,
+        errorMsgBody: undefined,
+        appUpdateButton: false,
+        allowFunction: false,
+        denyAccess: false,
+      }
+    }
+  }
+}
+
+export const loadWaygateOverrides = async (): Promise<void> => {
+  try {
+    const waygateOverrides = await AsyncStorage.getItem(WAYGATE_OVERRIDES_KEY)
+    if (waygateOverrides) {
+      setOverrideRemote(true)
+      waygateConfig = JSON.parse(waygateOverrides) as WaygateToggleValues
+    }
+  } catch (err) {
+    logNonFatalErrorToFirebase(err, 'loadOverrides: AsyncStorage error')
+  }
+}
+
+export const waygateNativeAlert = (feature: WaygateToggleType): boolean => {
+  const waygate = waygateEnabled(feature)
+  if (waygate.enabled === false && waygate.denyAccess === true && waygate.errorMsgTitle) {
+    Alert.alert(waygate.errorMsgTitle, waygate.errorMsgBody, [
+      {
+        text: 'OK',
+        style: 'cancel',
+      },
+    ])
+    return false
+  }
+  return true
+}
+
+export const setWaygateDebugConfig = async (config: WaygateToggleValues): Promise<void> => {
+  setOverrideRemote(true)
+  waygateConfig = config
+
+  // Store overrides in AsyncStorage so they persist with app quits
+  AsyncStorage.setItem(WAYGATE_OVERRIDES_KEY, JSON.stringify(config))
+}
+
+export const getWaygateToggles = (): WaygateToggleValues => {
+  if (overrideRemote) {
+    return waygateConfig
+  }
+
+  //this just initializes the waygate list without having to first create them all in firebase.
+  const toggles = waygateConfig
```

Comment:
I need more clarification on this one?

Reply:
This is just assigning `waygateConfig` to `toggles`, so toggles becomes a reference to waygateConfig. If you removed `toggles` and just performed the same operations on waygateConfig directly, the result would be the same. Maybe the intention here was to clone waygateConfig into toggles to avoid mutating waygateConfig, but it's not cloning so it is mutating